Problems with "wasm-type" when compiles with tot

Latest changes in `wasm-type.cpp` caused to failed compilation in binaryen.js:

```
In file included from /home/runner/work/binaryen.js/binaryen.js/binaryen/src/wasm/wasm-type.cpp:28:
/home/runner/work/binaryen.js/binaryen.js/binaryen/src/wasm-type.h:385:10: error: definition of implicit copy constructor for 'Tuple' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  Tuple& operator=(const Tuple&) = delete;
         ^
/home/runner/work/binaryen.js/binaryen.js/binaryen/src/wasm/wasm-type.cpp:60:51: note: in implicit copy constructor for 'wasm::Tuple' first required here
  TypeInfo(const Tuple& tuple) : kind(TupleKind), tuple(tuple) {}
                                                  ^
1 error generated.
em++: error: '/home/runner/emsdk/upstream/bin/clang++ -target wasm32-unknown-emscripten -DEMSCRIPTEN -fno-inline-functions -flto=full -D__EMSCRIPTEN_major__=2 -D__EMSCRIPTEN_minor__=0 -D__EMSCRIPTEN_tiny__=17 -D_LIBCPP_ABI_VERSION=2 -Dunix -D__unix -D__unix__ -Werror=implicit-function-declaration -Xclang -iwithsysroot/include/SDL --sysroot=/home/runner/emsdk/upstream/emscripten/cache/sysroot -Xclang -iwithsysroot/include/compat -I/home/runner/work/binaryen.js/binaryen.js/binaryen/src -I/home/runner/work/binaryen.js/binaryen.js/binaryen/build -std=c++14 -Wall -Werror -Wextra -Wno-unused-parameter -fno-omit-frame-pointer -fno-rtti -Wno-implicit-int-float-conversion -Wno-unknown-warning-option -Wswitch -Wimplicit-fallthrough -Wnon-virtual-dtor -O3 -DNDEBUG -Oz -UNDEBUG -flto -c /home/runner/work/binaryen.js/binaryen.js/binaryen/src/wasm/wasm-type.cpp -o CMakeFiles/wasm.dir/wasm-type.cpp.o' failed (1)
make[3]: *** [src/wasm/CMakeFiles/wasm.dir/build.make:204: src/wasm/CMakeFiles/wasm.dir/wasm-type.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:819: src/wasm/CMakeFiles/wasm.dir/all] Error 2
```

more details: https://github.com/AssemblyScript/binaryen.js/runs/2415417061?check_suite_focus=true#step:7:105

--- a/src/wasm-type.h
+++ b/src/wasm-type.h
@@ -377,12 +377,16 @@ struct Tuple {
   Tuple(std::initializer_list<Type> types) : types(types) { validate(); }
   Tuple(const TypeList& types) : types(types) { validate(); }
   Tuple(TypeList&& types) : types(std::move(types)) { validate(); }
+
+  // Allow copies when constructing.
+  Tuple(const Tuple& other) : types(other.types) { validate(); }
+
+  // Prevent accidental copies.
+  Tuple& operator=(const Tuple&) = delete;
+
   bool operator==(const Tuple& other) const { return types == other.types; }
   bool operator!=(const Tuple& other) const { return !(*this == other); }
   std::string toString() const;
-
-  // Prevent accidental copies
-  Tuple& operator=(const Tuple&) = delete;
 
 private:
   void validate() {

